### PR TITLE
Finished test

### PR DESCRIPTION
**Pull request by jenesh**

### CODE REVIEW

File: unitOneAssessment.js [hunk 1]
```diff
@@ -4,55 +4,74 @@ let assert = require('assert')
...
+  return outPut.join('');
+}
 // Uncomment out the next line to test your solution
-// runQ3Tests()
+runQ3Tests()
 
 // Question Four:
 // Write a function called secondSmallest that returns the second smallest number in an array
-
+const secondSmallest = (arr) => arr.sort((a,b) => a - b)[1]
```
Comment (alejo4373) [inline unitOneAssessment.js]: What's the time complexity of this? Can we improve it?

File: unitOneAssessment.js [hunk 2]
```diff
@@ -4,55 +4,74 @@ let assert = require('assert')
...
 // Question Three:
 
 // Write a function called disemvowel that removes all of the vowels from a string.
 // Treat y as a consonant, not a vowel
-
+const disemvowel = (str) => {
+  let strArr = str.split('');
+  let outPut = [];
+  for (let i = 0; i < strArr.length; i++) {
+    if (strArr[i] === 'a' || strArr[i] === 'A' ||strArr[i] === 'i' ||strArr[i] === 'I' ||strArr[i] === 'o' ||
```
Comment (alejo4373) [inline unitOneAssessment.js]: I `switch` statement would have been easier to read for a fellow developer. What would that look like?

File: unitOneAssessment.js [hunk 3]
```diff
@@ -4,55 +4,74 @@ let assert = require('assert')
...
-// runQ2Tests()
+runQ2Tests()
 
 // Question Three:
 
 // Write a function called disemvowel that removes all of the vowels from a string.
 // Treat y as a consonant, not a vowel
-
+const disemvowel = (str) => {
+  let strArr = str.split('');
```
Comment (alejo4373) [inline unitOneAssessment.js]: Why did you need to split?

File: unitOneAssessment.js (hunk 1 above)
Comment (MikePursuit) [reply unitOneAssessment.js]: O(n^2)

File: unitOneAssessment.js (hunk 2 above)
Comment (MikePursuit) [reply unitOneAssessment.js]: or (/[`a`|`e`|`i`|`o`|`u`]/ig)

File: unitOneAssessment.js (hunk 3 above)
Comment (MikePursuit) [reply unitOneAssessment.js]: split will allow us to use array methods like filter

File: unitOneAssessment.js (hunk 2 above)
Comment (jenesh) [reply unitOneAssessment.js]:
``` js
let outPut = '';
  for (let i = 0; i < str.length; i++) {
    switch (str[i]) {
      case 'a':
      case 'A':
      case 'e':
      case 'E':
      case 'i':
      case 'I':
      case 'o':
      case 'O':
      case 'u':
      case 'U':
        break;
      default:
        outPut += str[i]
      break;
    }
  }
  return outPut;
```

Comment (jenesh) [reply unitOneAssessment.js]: All of that would be inside of the function.

File: unitOneAssessment.js (hunk 1 above)
Comment (jenesh) [reply unitOneAssessment.js]: At worst it would be O(n^2) but should be less than that. I believe we could improve it with an O(n) without sorting using code below in a function: 
``` js
const secondSmallest = (arr) => {
  let min = Infinity;
  let sndMin = Infinity;
  arr.forEach(num => {
    if (num < min) {
      sndMin = min;
      min = num;
    } else if (num <sndMin) {
      sndMin = num;
    }
  })
  return sndMin;
}
```

File: unitOneAssessment.js (hunk 3 above)
Comment (jenesh) [reply unitOneAssessment.js]: I just like to split strings into arrays so I know they data type I'm working with is an array that way I can use array methods and manipulate arrays however I want. But for this question it isn't necessary we could just loop over string without turning it into an array.

File: unitOneAssessment.js (hunk 1 above)
Comment (alejo4373) [reply unitOneAssessment.js]: Yes, sorting in the worst case will be O(n<sup>2</sup>). Often times we can assume a reasonable implementation of Quicksort or Mergesort which will be O(n log n) on average. See [this](https://www.quora.com/What-is-the-sorting-algorithm-behind-a-Javascript-Array-sort-method)

File: unitOneAssessment.js (hunk 2 above)
Comment (alejo4373) [reply unitOneAssessment.js]: This is good. What if you used `str[i].toLowerCase()` in the `switch`?

Comment (alejo4373) [reply unitOneAssessment.js]: @MikePursuit how would you use the regex you suggested? It didn't work for me.

These are 2 ways I tried that work. Regexs are pretty cool. Both here are practically equivalent : 

```js
const disemvowel = (str) => str.replace(/[aeiou]/ig, '')
const disemvowel2 = (str) => str.replace(/a|e|i|o|u/ig, '')
```

Comment (alejo4373) [reply unitOneAssessment.js]: Was trying to think about the time complexity of this. I would say linear time and space but the time complexity of Regex is still obscure to me. I would suggest checking out [this article](https://mrale.ph/blog/2016/11/23/making-less-dart-faster.html) (I couldn't finish reading it)

File: unitOneAssessment.js (hunk 3 above)
Comment (alejo4373) [reply unitOneAssessment.js]: Correct! If you don't plan/need to take advantage of array properties like its methods no need to split
